v03_pipeline/lib/misc/vets.py

```diff
@@ -0,0 +1,40 @@
+import hail as hl
+
+PASS_FILTER = 'PASS'  # noqa: S105
+VETS_SNP_CUTOFF = 0.997
+VETS_INDEL_CUTOFF = 0.99
+VETS_SNP_FILTER = 'high_CALIBRATION_SENSITIVITY_SNP'
+VETS_INDEL_FILTER = 'high_CALIBRATION_SENSITIVITY_INDEL'
+
+
+def annotate_vets(mt: hl.MatrixTable) -> hl.MatrixTable:
+    if not hasattr(mt, 'info') or not hasattr(mt.info, 'CALIBRATION_SENSITIVITY'):
+        return mt
+    return mt.annotate_rows(
+        filters=hl.bind(
+            lambda is_snp, split_cs: (
+                hl.case()
+                .when(
+                    is_snp & (split_cs > VETS_SNP_CUTOFF),
+                    hl.if_else(
+                        hl.is_defined(mt.filters),
+                        mt.filters.add(VETS_SNP_FILTER).remove(PASS_FILTER),
+                        hl.set([VETS_SNP_FILTER]),
+                    ),
+                )
+                .when(
+                    ~is_snp & (split_cs > VETS_INDEL_CUTOFF),
+                    hl.if_else(
+                        hl.is_defined(mt.filters),
+                        mt.filters.add(VETS_INDEL_FILTER).remove(PASS_FILTER),
+                        hl.set([VETS_INDEL_FILTER]),
+                    ),
+                )
+                .default(
+                    mt.filters,
```

Comment:
do you need to add the `.remove(PASS_FILTER)` to this case as well?

Reply:
I don't think so!  that would remove `PASS` from the variants that do actually pass!

Reply:
GATK has an empty array for filters if they pass. There is no special handling in the hail backend if PASS is included, seqr will filter out variants as if they failed a filter if theres anything in the filter list

Reply:
ah, ok!  adding `.remove(PASS_FILTER)` was a complete mistake then.  It ended up in the tests as an example of an existing filter to preserve, but `PASS` isn't actually present in these vcfs.  I'll fix!